apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/Operation.kt

```diff
@@ -60,6 +61,34 @@ interface Operation<D : Operation.Data, T, V : Operation.Variables> {
   @Throws(IOException::class)
   fun parse(source: BufferedSource): Response<T>
 
+  /**
+   * Composes POST JSON-encoded request body with provided [scalarTypeAdapters] to be sent to the GraphQL server.
+   *
+   * *Example*:
+   * ```
+   * {
+   *    "query": "query TestQuery($episode: Episode) { hero(episode: $episode) { name } }",
+   *    "operationName": "TestQuery",
+   *    "variables": { "episode": "JEDI" }
+   * }
+   * ```
+   */
+  fun composeRequestBody(scalarTypeAdapters: ScalarTypeAdapters): ByteString
```

Comment:
Just curious: what's the advantage of  exposing a `ByteString` instead of a `String` ? Do we support something else than utf8 ?

Reply:
Sure we can make it as a `String`, I was optimizing for `okhttp3.RequestBody`
https://github.com/square/okhttp/blob/master/okhttp/src/main/kotlin/okhttp3/RequestBody.kt#L105

in case of string it will convert it to back to byte array, so we are saving here extra conversions `bytestring -> string -> bytestring`.